emoji/src/main/java/com/vanniktech/emoji/EmojiManager.java

```diff
@@ -13,9 +20,18 @@
  */
 public final class EmojiManager {
   private static final EmojiManager INSTANCE = new EmojiManager();
+  private static final int GUESSED_UNICODE_AMOUNT = 3000;
+  private static final int GUESSED_TOTAL_PATTERN_LENGTH = GUESSED_UNICODE_AMOUNT * 4;
 
-  private final EmojiTree emojiTree = new EmojiTree();
+  private static final Comparator<String> STRING_LENGTH_COMPARATOR = new Comparator<String>() {
+    @Override public int compare(final String first, final String second) {
+      return second.length() - first.length();
```

Comment:
This can cause issues. Have a look at [this](https://stackoverflow.com/a/2728858/1979703). It's an extreme edge case though, although we should cover it.